### ISSUE TITLE
[dev.icinga.com #5593] Attribute authorities ignored by Cluster Master Node

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#1212]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/5593

**Created by ccesario on 2014-02-03 19:32:23 +00:00**

Assignee: _gbeutner_
Status: _Resolved_  (closed on _2014-02-10 09:53:41 +00:00_)
Target Version: _0.0.8_
Last Update: _2014-02-10 09:53:41 +00:00 (in Redmine)_

    Icinga Version: Icinga 2 - 0.0.7

---

Hello, a enable **checker** module on master cluster node some hosts/services get down/offile. Investigating the log file/state file I got that when I enable checker module on the master cluster its ignore the authorities attribute defined in host config.

**Icinga state partial file**

    "last_check_result":    {
                "__type":   "CheckResult",
                "active":   1,
                "check_source": "icinga-node-1",
                "command":  ["/usr/lib/nagios/plugins/check_ping", "-4", "-H", "192.168.0.5", "-w", "100,5%", "-c", "200,15%", "-p", "5", "-t", "0"],
                "execution_end":    1.391441461922437e+09,
                "execution_start":  1.391441455920221e+09,
                "exit_status":  2,
                "output":   "PING CRITICAL - Packet loss = 100%",
                "schedule_end": 1.391441461922536e+09,
                "schedule_start":   1.391441455920000e+09,
                "state":    2,
                "vars_after":   {

**Host configuration**

    /**
     * A host definition. You can create your own configuration files
     * in the conf.d directory (e.g. one per host). By default all *.conf
     * files in this directory are included.
     */
    object Host "Host::Firewall" inherits "linux-server" {
      display_name = "host::firewall",
      authorities = [ "icinga-node-3" ],

      services["ping4"] = {
        templates = [ "generic-service" ],

        check_command = "ping4"
      },

      services["http"] = {
        templates = [ "generic-service" ],

        check_command = "http_ip"
      },

      services["ssh"] = {
        templates = [ "generic-service" ],

        check_command = "ssh"
      },

      macros = {
        address = "192.168.0.5",
      },

      check = "ping4",
    }

As the state file show the, [check\_source]() is "**icinga-node-1**" ... and the correct is **icinga-node-3** as defined in host config authorities = \[ "icinga-node-3" \].

If I disable the checker module on master cluster its work as expected!

**Changesets**

_2014-02-10 07:50:09 +00:00_ by _gbeutner_ 663f72300be279a1ed4feedf61e722ea1f1f74dd

    Fix validation for the "authorities" attribute.
    
    Fixes #5593
